src/storage/src/hummock/compactor.rs

```diff
@@ -147,10 +147,59 @@ impl Compactor {
         }
     }
 
+    /// Flush shared buffer to level0. Resulted SSTs are grouped by compaction group.
+    pub async fn compact_shared_buffer_by_compaction_group(
+        context: Arc<CompactorContext>,
+        payload: UploadTaskPayload,
+    ) -> HummockResult<Vec<(CompactionGroupId, Sstable, u64, Vec<VNodeBitmap>)>> {
+        let mut grouped_payload: HashMap<CompactionGroupId, UploadTaskPayload> = HashMap::new();
+        for uncommitted_list in payload {
+            let mut next_inner = HashSet::new();
+            for uncommitted in uncommitted_list {
+                let batch = match &uncommitted {
+                    UncommittedData::Sst(_) => {
+                        panic!("expect UncommittedData::Batch")
```

Comment:
Currently in when we compact the shared buffer, for the compact task issued from sync call, we may still include sst. So we cannot panic here.

Reply:
Fixed